### PR TITLE
Add last execution by group

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR adds an API endpoint to retrieve the last execution by `group_id`.
We need this in the FE to get the last executions in the components instead of fetching a list with an artificial pagination.

### CODE REVIEW

**Review by CDimonaco — APPROVED** [review]

File: test/wanda_web/controllers/execution_controller_test.exs [hunk 1]
```diff
@@ -67,6 +67,26 @@ defmodule WandaWeb.ExecutionControllerTest do
...
+        conn
+        |> get("/api/checks/groups/#{group_id}/executions/last")
+        |> json_response(200)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "ExecutionResponse", api_spec)
+    end
+
+    test "should return a 404", %{conn: conn} do
+      assert_error_sent 404, fn ->
```
Comment (arbulu89) [inline test/wanda_web/controllers/execution_controller_test.exs:84]: Question. How does this even know that it should return 404?
Does elixir capture the specific error from Ecto and is able to know that is not found?

File: lib/wanda_web/router.ex [hunk 2]
```diff
@@ -12,6 +12,7 @@ defmodule WandaWeb.Router do
     pipe_through :api
 
     resources "/executions", ExecutionController, only: [:index, :show]
+    get "/groups/:id/executions/last", ExecutionController, :last
```
Comment (arbulu89) [inline lib/wanda_web/router.ex:15]: I guess this removes the need to have the `group_id` filter in the `/executions` endpoint, as we would maybe better have `/groups/:id/executions`.
Anyway, let's not bother now. Both would do the same and 1st is already implemented

**Review by arbulu89 — APPROVED** [review]

**Review by nelsonkopliku — APPROVED** [review]

**Review by dottorblaster — APPROVED** [review]
LGTM

File: test/wanda_web/controllers/execution_controller_test.exs (hunk 1 above)
Comment (fabriziosestito) [reply test/wanda_web/controllers/execution_controller_test.exs:84]: `Ecto.NoResultsError` exception is caught by a Plug that displays a 404, this is provided by Phoenix.
In development you'll see the phoenix debug page for every exception including the 404,  in production you'll just get a 404 error view.

File: lib/wanda_web/router.ex (hunk 2 above)
Comment (fabriziosestito) [reply lib/wanda_web/router.ex:15]: the group_id filter could be still  helpfpul when we'll do the historical result view

**Pull request merged**